src/Native/System.Private.CoreLib.Native/configure.cmake

```diff
@@ -60,6 +62,20 @@ int main(void)
 }" HAVE_SCHED_GETCPU)
 set(CMAKE_REQUIRED_LIBRARIES)
 
+if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
```

Comment:
If a special handling for FreeBSD is needed for some reason, it could be just:

```
if(CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
  # Put special FreeBSD handing here
else()
  if(NOT HAVE_LIBUUID_H) 
    unset(HAVE_LIBUUID_H CACHE)
    message(FATAL_ERROR "Cannot find libuuid. Try installing uuid-dev or the appropriate packages for your platform") 
  endif()
endif()
```

However, I think it would be better to just keep this as simple as possible, and add the extra stuff once somebody can test it on FreeBSD.